### ISSUE TITLE
Ability to show labels on far-left

### ISSUE THREAD

**Issue by peppy** [mhutchie/vscode-git-graph#88]
Thanks for the amazing work you're doing on git graph. I'm eager to get to the point I can switch over from gitkraken, but I have to say that I still prefer branch labels displayed to the left of the graph (while tags are in the middle or right aligned):

![image](https://user-images.githubusercontent.com/191335/58530149-9c9ef580-8218-11e9-92f9-81a56c715f24.png)

I've found gitkraken to be one of the easiest to read git graph displays and I think a few more customisation options for layout could get git-graph on par!

**Comment by mhutchie** [comment]
Thanks for the suggestion! I'll add this in an upcoming release.

**Comment by andreiucm** [comment]
@mhutchie that will increase the readability a lot for me. Will be nice to have that feature

**Comment by tuc0w** [comment]
@mhutchie is there any progress on this feature? I am migrating from GitKraken and this is _THE_ feature I am missing.

Thanks for your work on this extension!

**Comment by mhutchie** [comment]
Hi @tuc0w,

I’ve explored and done some POC’s around how this functionality would be implemented in Git Graph, and found that there a variety of challenges that require a significant amount of work in order to produce a really nice solution that is flexible to a wide range of user workflows.

Considering only three people (including yourself) out of 297,000 Git Graph users have indicated they would like this feature, and the significant implementation complexity, it is relatively low priority to be implemented in Git Graph.

I know changing between or evaluating new Git UI’s can be challenging considering you are used to how your previous Git UI functions. Thank you for taking the time to try out Git Graph, I hope you continue to give it a go and see what works best for you and your development workflow.

**Comment by tuc0w** [comment]
Hey @mhutchie, 

thank you for your answer and explaining why it is low priority.

Now it is quite understandable why there's no "progress" on this, I think I'll continue using it but also trying to implement it myself :)

**Comment by mhutchie** [comment]
Hi @tuc0w,

These are just some of the challenges that need to be addressed if you have a go at implementing this yourself:
* The graph (which rendered separately from the table containing the commits) needs to automatically adjust its horizontal position based on the widths of the reference labels on the left.
* Handle many labels on a single commit - a combination of intuitive abbreviations of the labels and overflow pop ups so that the graph doesn’t move so far to the right that there isn’t enough room for the text in the “Description” column to be useful.
* Use width efficiently to support users with extremely large repositories (many users work on these repos from the feedback I’ve gotten). By large I mean 20+ branch column in the graph and 8+ reference labels on some commits.
* If the user manually resizes the graph table column, the labels need to collapse at an appropriate rate to the graph itself, to respect the width set by the user.

If you work out suitable solutions to all of the above challenges, please let me know so I can provide some other less important challenges that also need to be addressed before this functionality can be released.

It would also need to be included as a new option in the extension setting controlling the reference label alignment, however this is trivial to do.

Other content you should read and be aware of before starting work (that people often disregard):
* [Contributing to Development](https://github.com/mhutchie/vscode-git-graph/blob/master/CONTRIBUTING.md#contributing-to-development) - I’ve provided a set of steps to help get anyone ready to develop Git Graph.
* [License](https://github.com/mhutchie/vscode-git-graph/blob/master/LICENSE)

**Comment by mhutchie** [comment]
This issue has been left open for 13 months for others to express interest in this feature. As only two other Git Graph users have indicated they would like this functionality in this period, it seems to confirm my previous sentiment that this feature wouldn't be used by many users.

Although this functionality would definitely be nice to have, and would be valued by those who use it, because of the complex & time consuming implementation challenges I previously mentioned, and that few other users have indicated they would like this feature, I'll close it for now and move it to the backlog.

Note to users reading this in the future: If you'd like this functionality, please reply to this thread to express your interest. I'd re-consider implementing it if it would be used by a decent number of users.

**Comment by e-compton-bb** [comment]
I'd also be keen on this feature. Have been digging through the settings trying to find a way to have the labels aligned with each other but separate from commit descriptions so that I can quickly orient myself
